File: app/src/main/java/info/firozansari/animatedvectordrawable/ExampleActivity.java
package info.firozansari.animatedvectordrawable;

import android.app.Activity;
import android.graphics.drawable.Animatable;
import android.graphics.drawable.Drawable;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

public class ExampleActivity extends AppCompatActivity {
  private ViewGroup container;

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_example);

    container = (ViewGroup) findViewById(R.id.container);
    container.setOnClickListener(new View.OnClickListener() {
      @Override
      public void onClick(View view) {
        for (int i = 0; i < container.getChildCount(); ++i) {
          animateDrawables(container.getChildAt(i));
        }
      }
    });
    for (int i = 0; i < container.getChildCount(); ++i) {
      View child = container.getChildAt(i);
      if (child instanceof TextView) {
        child.setOnClickListener(new View.OnClickListener() {
          @Override
          public void onClick(View view) {
            for (int i = 0; i < container.getChildCount(); ++i) {
              animateDrawables(container.getChildAt(i));
            }
          }
        });
      }
    }
  }

  private void animateDrawables(View view) {
    if (!(view instanceof TextView)) {
      return;
    }
    TextView textView = (TextView) view;
    for (final Drawable drawable : textView.getCompoundDrawables()) {
      if (drawable instanceof Animatable) {
        ((Animatable) drawable).start();
      }
    }
  }
}

File: app/src/main/java/info/firozansari/animatedvectordrawable/ExampleActivity.kt
package info.firozansari.animatedvectordrawable

import android.graphics.drawable.Animatable
import android.os.Bundle
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class ExampleActivity : AppCompatActivity() {
    private var container: ViewGroup? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_example)
        container = findViewById<ViewGroup>(R.id.container).setOnClickListener({
            for (i in 0 until container.getChildCount()) {
                animateDrawables(container.getChildAt(i))
            }
        })

        for (i in 0 until container.getChildCount()) {
            val child: View = container.getChildAt(i)
            (child as? TextView)?.setOnClickListener(View.OnClickListener {
                for (i in 0 until container.getChildCount()) {
                    animateDrawables(container.getChildAt(i))
                }
            })
        }
    }

    private fun animateDrawables(view: View) {
        if (view !is TextView) {
            return
        }
        val textView: TextView = view as TextView
        for (drawable in textView.getCompoundDrawables()) {
            if (drawable is Animatable) {
                (drawable as Animatable).start()
            }
        }
    }
}
